src/forward_sensitivity.jl

```diff
@@ -218,9 +218,22 @@ has_continuous_callback(cb::CallbackSet) = !isempty(cb.continuous_callbacks)
 
 function ODEForwardSensitivityProblem(f::DiffEqBase.AbstractODEFunction,u0,
                                       tspan,p,alg::ForwardDiffSensitivity;
+                                      du0=zeros(eltype(u0),length(u0),length(p)), # perturbations of initial condition
+                                      dp=I(length(p)), # perturbations of parameters
                                       kwargs...)
-  pdual = seed_duals(p,f)
-  u0dual = convert.(eltype(pdual),u0)
+  num_sen_par = size(du0,2)
+  if num_sen_par != size(dp,2)
+    error("Same number of perturbations of initial conditions and parameters required")
+  end
+  if size(du0,1) != length(u0)
+    error("Perturbations for all initial conditions required")
+  end
+  if size(dp,1) != length(p)
+    error("Perturbations for all parameters required")
+  end  
+
+  pdual = ForwardDiff.Dual{typeof(ForwardDiff.Tag(f,eltype(vec(p))))}.(p, [ntuple(j -> dp[i,j], num_sen_par) for i in eachindex(p)])
+  u0dual = ForwardDiff.Dual{typeof(ForwardDiff.Tag(f,eltype(vec(u0))))}.(u0, [ntuple(j -> du0[i,j], num_sen_par) for i in eachindex(u0)])
```

Comment:
This will make everything not infer 😅 . What happens to the compile times?

Reply:
I don't think it was inferring properly before?

```jl
using Pkg; Pkg.activate("./old")
Pkg.add("OrdinaryDiffEq")
@time Pkg.add("DiffEqSensitivity") # 82.320349 seconds (4.59 M allocations: 459.409 MiB, 0.07% gc time, 0.29% compilation time)
using OrdinaryDiffEq
@time using DiffEqSensitivity # 14.851568 seconds (59.35 M allocations: 4.020 GiB, 3.75% gc time, 33.46% compilation time)

function f(du, u, p, t)
    du[1] = dx = p[1] * u[1] - p[2] * u[1] * u[2]
    du[2] = dy = -p[3] * u[2] + u[1] * u[2]
end
u0 = [1.0,1.0]
p = [1.5,1.0,3.0]
tspan = (0.0,10.0)

@time prob = ODEForwardSensitivityProblem(f,u0,tspan,p,ForwardDiffSensitivity()); # 1.946009 seconds (6.18 M allocations: 519.725 MiB, 2.98% gc time, 99.68% compilation time)
@time sol = solve(prob,Tsit5(),reltol=1e-14) # 2.908536 seconds (11.21 M allocations: 721.006 MiB, 3.86% gc time, 99.98% compilation time)
```

```jl
using Pkg; Pkg.activate("./new")
Pkg.add("OrdinaryDiffEq")
@time Pkg.add(url="https://github.com/ArnoStrouwen/DiffEqSensitivity.jl", rev="patch-1") #  18.263005 seconds (5.31 M allocations: 434.693 MiB, 0.24% gc time, 2.76% compilation time)
using OrdinaryDiffEq 
@time using DiffEqSensitivity # 15.177978 seconds (59.97 M allocations: 4.059 GiB, 5.04% gc time, 35.26% compilation time)

function f(du, u, p, t)
    du[1] = dx = p[1] * u[1] - p[2] * u[1] * u[2]
    du[2] = dy = -p[3] * u[2] + u[1] * u[2]
end
u0 = [1.0,1.0]
p = [1.5,1.0,3.0]
tspan = (0.0,10.0)

@time prob = ODEForwardSensitivityProblem(f,u0,tspan,p,ForwardDiffSensitivity()); # 0.539928 seconds (2.16 M allocations: 121.213 MiB, 2.48% gc time, 98.71% compilation time)
@time sol = solve(prob,Tsit5(),reltol=1e-14) # 2.887392 seconds (11.44 M allocations: 735.403 MiB, 3.81% gc time, 99.98% compilation time)

```